src/server/pkg/log/log.go

```diff
@@ -37,6 +73,113 @@ func (l *logger) Log(request interface{}, response interface{}, err error, durat
 	} else {
 		l.LogAtLevelFromDepth(request, response, err, duration, logrus.InfoLevel, 4)
 	}
+	// We have to grab the method's name here before we
+	// enter the goro's stack
+	go l.ReportMetric(getMethodName(), duration, err)
+}
+
+func getMethodName() string {
+	depth := 4
+	pc := make([]uintptr, depth)
+	runtime.Callers(depth, pc)
+	split := strings.Split(runtime.FuncForPC(pc[0]).Name(), ".")
+	return split[len(split)-1]
+}
+
+func (l *logger) ReportMetric(method string, duration time.Duration, err error) {
+	// Count the number of ReportMetric() goros in case we start to leak them
+	if reportMetricGauge != nil {
```

Comment:
I think you can probably remove this `nil` check, it shouldn't be possible for this method to be called without `NewLogger` having already returned and thus the Gauge having been set.

Reply:
So, I don't initialize the value of `reportMetricGauge` unless the registration of that gauge w prometheus succeeds. If it doesn't, it will be nil. That seemed like the easiest way to degrade in case registration errs

Reply:
Ahh, I see. Makes sense then. Just for my own knowledge, do you have a sense of what could cause registration with Prometheus to fail and how common that's going to be?